src/StaticWebAssetsSdk/Tasks/ApplyCompressionNegotiation.cs

```diff
@@ -89,13 +89,14 @@ public override bool Execute()
                 }
 
                 Log.LogMessage(MessageImportance.Low, "  Updated endpoint '{0}' with Content-Encoding and Vary headers", compressedEndpoint.Route);
-                if (!updatedEndpoints.Contains(compressedEndpoint))
-                {
-                    updatedEndpoints.Add(compressedEndpoint);
-                }
+                updatedEndpoints.Add(compressedEndpoint);
 
                 foreach (var relatedEndpointCandidate in relatedAssetEndpoints)
                 {
+                    if (!IsCompatible(compressedEndpoint, relatedEndpointCandidate))
+                    {
+                        continue;
+                    }
```

Comment:
We were copying the related endpoint and applying a selector + the compressed endpoint properties. That caused for example that we created a fingerprinted endpoint using a non-fingerprinted compressed endpoint. 

To address that, we check that both endpoints have or don't have fingerprint and in case of having it, they match. 

This way the fingerprinted endpoint we create for content-negotiation has the same headers as the fingerprinted compressed endpoint.